group_vars/all.yml

```diff
@@ -196,7 +196,8 @@ nginx_sites_available:
       location /products.json {
         proxy_cache public_cache;
         add_header X-Cache-Status $upstream_cache_status;
-        proxy_ignore_headers Set-Cookie;
+        proxy_ignore_headers Set-Cookie; # ignore cookie headers to enable caching
+        proxy_hide_header Set-Cookie; # strip cookie header to ensure it's not cached and shared
```

Comment:
I guess that we would need to modify our Ajax call to not set a cookie and tell Rails not to set a cookie. That's more complicated but it would be the better option. Many platforms choose a separate domain for a CDN to avoid the cookie management.

So the other option would be to turn this file into an asset that is compiled and uploaded to CDN every five minutes.